snakemake/deployment/__init__.py

```
@@ -0,0 +1,103 @@
+                    f"The software deployment plugin {plugin.name} provides an "
+                    f"invalid kind {kind}, which is already reserved as a global "
+                    "variable in snakemake."
+                )
+
+        for kind, plugin in self.plugins.items():
+            def make_factory(plugin_val, kind_val):
+                def factory(*args, within: EnvSpecBase, **kwargs):
+                    if args:
+                        raise WorkflowError(
+                            "Positional arguments are not allowed in "
+                            f"software deployment expressions (found in {kind}())"
```

Ruff (0.8.2)

Function definition does not bind loop variable `kind`

(B023)